tests/test_lcov.py

```diff
@@ -0,0 +1,148 @@
+            return line_list
+
+    def test_no_lcov_at_start(self):
+        """Preliminary test to show that there is no lcov file at start"""
+        self.assert_doesnt_exist("coverage.lcov")
+
+    def test_lcov_file_exists(self):
+        """Test that the coverage.lcov file is produced when coverage is run"""
+        self.create_initial_files()
+        cov = self.run_coverage()
+        cov.lcov_report()
+        self.assert_exists("coverage.lcov")
+
+    def test_lone_file(self):
+        """For a single file with a couple of functions, the lcov should cover
+        the function definitions themselves, but not the returns"""
+        self.make_file(
+            "main_file.py",
+            """\
+            #!/usr/bin/env python3
+
+            def cuboid_volume(l):
+                return (l*l*l)
+
+            def IsItTrue():
+                return True
+
+        """,
+        )
+        expected_result = [
+            "TN:\n",
+            "SF:main_file.py\n",
+            "DA:3,1,7URou3io0zReBkk69lEb/Q\n",
+            "DA:6,1,ilhb4KUfytxtEuClijZPlQ\n",
+            "DA:4,0,Xqj6H1iz/nsARMCAbE90ng\n",
+            "DA:7,0,LWILTcvARcydjFFyo9qM0A\n",
+            "LF:4\n",
+            "LH:2\n",
+            "end_of_record\n",
+        ]
+        cov = coverage.Coverage(source=["."])
+        self.start_import_stop(cov, "main_file")
+        cov.lcov_report()
+        actual_result = self.get_lcov_report_content()
+        assert expected_result == actual_result
+
+    def test_simple_line_coverage_two_files(self):
+        """Test that line coverage is created when coverage is run,
+        and matches the output of the file below"""
+        self.create_initial_files()
+        cov = coverage.Coverage(source=["."])
+        self.start_import_stop(cov, "main_file")
+        self.start_import_stop(cov, "test_file")
+        cov.lcov_report()
+        self.assert_exists("coverage.lcov")
+        expected_result = [
+            "TN:\n",
+            "SF:main_file.py\n",
+            "DA:3,1,7URou3io0zReBkk69lEb/Q\n",
+            "DA:6,1,ilhb4KUfytxtEuClijZPlQ\n",
+            "DA:4,0,Xqj6H1iz/nsARMCAbE90ng\n",
+            "DA:7,0,LWILTcvARcydjFFyo9qM0A\n",
+            "LF:4\n",
+            "LH:2\n",
+            "end_of_record\n",
+            "TN:\n",
+            "SF:test_file.py\n",
+            "DA:3,1,R5Rb4IzmjKRgY/vFFc1TRg\n",
+            "DA:4,1,E/tvV9JPVDhEcTCkgrwOFw\n",
+            "DA:6,1,GP08LPBYJq8EzYveHJy2qA\n",
+            "DA:7,1,MV+jSLi6PFEl+WatEAptog\n",
+            "DA:8,0,qyqd1mF289dg6oQAQHA+gQ\n",
+            "DA:9,0,nmEYd5F1KrxemgC9iVjlqg\n",
+            "DA:10,0,jodMK26WYDizOO1C7ekBbg\n",
+            "DA:11,0,LtxfKehkX8o4KvC5GnN52g\n",
+            "LF:8\n",
+            "LH:4\n",
+            "end_of_record\n",
+        ]
+        actual_result = self.get_lcov_report_content()
+        assert expected_result == actual_result
+
+    # TODO: Add tests for branch coverage here for when this
```

Comment:
Hmm, I thought the issue was that you didn't think this was possible with coveragepy because none of the other examples tried it?

It's worth noting that lcov [has two different kinds of branch coverage reporting](https://github.com/linux-test-project/lcov/blob/master/man/geninfo.1#L552):

1. Per-line "is this branch covered"
2. Summary of branch coverage

Having had a quick glance, the latter shouldn't be all that hard to do, see https://github.com/nedbat/coveragepy/blob/1c29ef3e4b871508bb2defd6b47b9a745547b626/coverage/results.py#L139 for getting the data.

I think you'd simply need to add something like:

```python
brf, brh = analysis.branch_stats()
outfile.write(f"BRF:{brf}\n")
outfile.write(f"BRH:{brh}\n")```

The latter can be done with [`arcs`](https://github.com/nedbat/coveragepy/blob/1c29ef3e4b871508bb2defd6b47b9a745547b626/coverage/results.py#L79), I believe, but I don't know enough about what those represent to actually recommend anything here without playing a bit with it myself.